components/input-number.tsx

```diff
@@ -58,7 +70,7 @@ export default function InputNumber({
           {helperText}
         </div>
       )}
-      {error && (
+      {error && error != ' ' && (
```

Comment:
this is a mighty odd condition. I know what's going on because I just read the code and thus I know what this is. ... a week from now? I'll think this is a bug.

Please, if you're short-circuiting logic use readable names or helping variables.

This means, either introduce a new "shortCircuitError: boolean" variable that's set to true or false when needed and use it here _or_ use a error-code-value to make the logical statement.